### PR TITLE
MBS-11995: Test error messages in external links editor

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11995

We should ensure we are returning the expected errors for tests that we expect to fail, including both specific message (when given) and target.
This just uses a substring, rather than the whole error message, because it seems like overkill otherwise and it's easier for oneswe expand with arguments.

Adds a new optional test property `expected_error`. This requires one of `input_relationship_type` or `expected_relationship_type` (since the error might vary depending on the relationship selected) and `input_entity_type` (since the same URL might be allowed for one entity type but not another).

This adds two examples:

One for our block of Wikipedia pages as discography entries for releases, testing a specific error message and `input_relationship_type`.

One for blocking non-specific 45cat pages, testing a generic error message (undefined error) and `expected_relationship_type`.

Further tests to cover other existing blocks and errors will be added in subsequent commits.

### CODE REVIEW

**Comment by reosarevok** [comment]
@yvanzo: can you check whether this seems sensible? If so, I can add further errors to tests.

**Review by yvanzo — COMMENTED** [review]
It looks good so far :+1: Testing with only one input relationship type seems to be sufficient as the error message for selecting an invalid combination of two or more relationship types is generic.

**Comment by reosarevok** [comment]
@yvanzo: found some issues while trying to expand this (it did not support testing for errors where a custom message isn't given). Added a second example case for those, and allowed using `expected_relationship_type` for cases where we do autoselect but then reject. I'd probably merge this as-is (if it seems good) and then expand its usage further, if that seems sensible?

File: root/static/scripts/tests/Control/URLCleanup.js
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:5329]: It seems that the flag `tested` should be raised after error has been tested.

File: root/static/scripts/tests/Control/URLCleanup.js
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:5328]: I suggest to move this part of the code to a separate function to improve readability.
(The same should have been done with validation tests really.)
Can be done after merging this pull request if in a hurry.

Comment (reosarevok) [reply root/static/scripts/tests/Control/URLCleanup.js]: I made the change in a second commit, but do let me know if that looks like what you expected.

**Review by yvanzo — APPROVED** [review]

**Pull request merged**
